hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallLocalHandler.java

```diff
@@ -74,6 +83,8 @@ public ResponseCodeEnum validate(@NonNull final Query query) throws PreCheckExce
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public Response findResponse(@NonNull final Query query, @NonNull final ResponseHeader header) {
+        Objects.requireNonNull(query);
```

Comment:
Q: are these needed or are both needed? Or what's the purpose of the `@NonNull` checks in the params signature.

Reply:
The annotation is only used by Sonar and similar tools, it doesn't exist at runtime.
The call to `Objects.requireNonNull()` is the runtime enforcement.